filesystem/squashfs_superblock.ksy

```diff
@@ -0,0 +1,116 @@
+meta:
+  id: squashfs_superblock
+  title: SquashFS superblock
+  file-extension:
+    - squashfs
+    - snap
+  endian: le
+  license: CC0-1.0
+  xref:
+    wikidata: Q389314
+doc: |
+  SquashFS is a compressed filesystem in a file, that can be read-only
+  accessed from low memory devices. It is popular for booting LiveCDs and
+  packing self-contained binaries. SquashFS format is used by Ubuntu .snap
+  packages. SquashFS is natively supported by Linux Kernel.
+doc-ref: https://github.com/AgentD/squashfs-tools-ng/blob/master/doc/format.txt
```

Comment:
Should link to a specific revision, not to `master` that can change over time:

```suggestion
doc-ref: https://github.com/AgentD/squashfs-tools-ng/blob/b76eae1/doc/format.txt
```

Reply:
Why not to the original squashfs project as well?

https://github.com/plougher/squashfs-tools/

Reply:
I guess because https://github.com/plougher/squashfs-tools/blob/master/COPYING

Reply:
> I guess because https://github.com/plougher/squashfs-tools/blob/master/COPYING

Again, this is not a problem. Also, for interoperability you are allowed to look at this in many jurisdictions without any problem at all.

Reply:
Depends heavily on court decisions on per-case basis. And I don't think that in the case where source code is available and when one explicitly claims he bases his work on that code, the court gonna decide that the work is independent and not derivative one. 

Copyright trolls are copyright trolls, and choosing GPL when not forced to, is one of the markers of a copyright troll.

>Also, for interoperability you are allowed to look at this in many jurisdictions without any problem at all.

It is in the jurisdictions where reversing for interoperability is fair use ... There are ones in which there is no concepts of fair use, but reversing is allowed ... with a lot of limitations .... only for the cases almost never happening in practice.

Reply:
> Depends heavily on court decisions on per-case basis. And I don't think that in the case where source code is available and when one explicitly claims he bases his work on that code, the court gonna decide that the work is independent and not derivative one.

I would find it very hard to believe that a header file that contains no code, just definitions and a description of an on disk structure, would be copyrightable.
 
> Copyright trolls are copyright trolls, and choosing GPL when not forced to, is one of the markers of a copyright troll.

This doesn't make sense. Who is choosing GPL here when not forced to?

Reply:
>Who is choosing GPL here when not forced to?

Not "here", but in general. Some people prefer to license their software to GPL and other licenses designed to create legal troubles to other people. Not everyone considers it as OK. In the current legal framework the best way to deal with such software and people is to avoid them.

Reply:
I vehemently disagree. I suggest take this discussion offline as to not clutter the comments :)

Reply:
@adrianherrera because https://github.com/plougher/squashfs-tools/ is not a link to reference doc.